File: src/main/kotlin/Popstack.kt
/**
 * This file is part of the PopStack (Kotlin implementation).
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2016 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

package popstack;

import popstack.stackoverflow.StackOverflowProvider;

/*TODO:
 * build consolidated jar (shade plugin)
 * maven project setup
 * code style
 * static code analysis
 * unit tests
 * auto documentation
 * logs
 */

fun main(args: Array<String>): Unit {
    val query = args.joinToString(" ");
    val provider = StackOverflowProvider();

    try {
        val answer = provider.search(query);

        if (answer != null) {
            println(answer);
        } else {
            println("Your only help is http://google.com/ man!");
        }
    } catch (error: Exception) {
        println(error);
    }
}

File: src/popstack.ts
/**
 * This file is part of the PopStack (TypeScript implementation).
 *
 * @license http://mit-license.org/ The MIT license
 * @copyright 2016 © by Rafał Wrzeszcz - Wrzasq.pl.
 */

/* TODO:
 * static code analysis
 * unit tests
 * auto documentation
 * logs
 */

import ProviderInterface from "./PopStack/ProviderInterface";
import StackOverflowProvider from "./PopStack/StackOverflow/StackOverflowProvider";

let query: string = process.argv.slice(2).join(" ");

let provider: ProviderInterface = new StackOverflowProvider();
provider.search(query)
    .then(function(answer: string): void {
            if (answer) {
                console.log(answer);
            } else {
                console.log("Your only help is http://google.com/ man!");
            }
    })
    .catch(function(error: string): void {
            console.log(error);
    });
